psi4/src/psi4/libmints/matrix.cc

```diff
@@ -404,6 +404,43 @@ void Matrix::copy(const Matrix &cp) { copy(&cp); }
 
 void Matrix::copy(const SharedMatrix &cp) { copy(cp.get()); }
 
+// produces an Eigen::Map object mapping to the matrix data buffer 
+// of a matrix with a single irrep 
+Eigen::Map<Eigen::MatrixXd> Matrix::eigen_map() {
+    // this function only works with matrices with a single irrep 
+    if (nirrep() != 1) {
+        std::string message = "Matrix::eigen_map() called, but matrix only has one irrep! ";
+        message += "Use Matrix::eigen_maps() instead."; 
+        
+        throw PSIEXCEPTION(message); 
+    }
+
+    // create Eigen matrix "map" using Psi4 matrix data array directly
+    return std::move(
+        Eigen::Map<Eigen::MatrixXd>(
+            get_pointer(), nrow(), ncol()
+        )
+    );
+}
+
+// produces Eigen::Maps object mapping to the matrix data buffer
+// of a matrix with multiple irreps 
+// NOTE: this impl for mapping Psi4 matrices to Eigen maps
+// is currently experimental, as it is unused in the code 
+// currently
+std::vector<Eigen::Map<Eigen::MatrixXd>> Matrix::eigen_maps() {
+    std::vector<Eigen::Map<Eigen::MatrixXd>> eigen_maps;
+    eigen_maps.reserve(nirrep());
```

Comment:
Why not pass `nirrep()` to the constructor on the line above.

Reply:
`Eigen::Map<Eigen::MatrixXd>` doesn't seem have a constructor definition for the case with no input arguments, i.e., `Eigen::Map<Eigen::MatrixXd>()`. Trying to do 

```std::vector<Eigen::Map<Eigen::MatrixXd>>(nirrep())``` 

will attempt to create a series of `Eigen::Map` objects using that constructor which, because it's undefined, causes compilation to fail. 

The way I implemented it gets around that by using `emplace_back` to construct each `Eigen::Map` in place, with the proper arguments, and add it to the overall `std::vector` upon construction.